### ISSUE TITLE
Is the node down virtual machine automatically migrated?

### ISSUE THREAD

**Issue by am6737** [kubevirt/kubevirt#10146]
**Is your feature request related to a problem? Please describe**:
After the VM node is shut down, the virt-launcher-vm state changes to Terminating.

There was no launch of a new virt-launcher-vm as with live migration.

I want to know if the virtual machine is automatically migrated to the new node when the node fails?

**Describe the solution you'd like**:

**Describe alternatives you've considered**:

**Additional context**:

**Comment by oshoval** [comment]
Hi am6737,

There is no automatic migration.

Only k8s itself manage its own resources and makes sure they are available even when node is down,
kubevirt doesn't do as such.

Migration should be initiated by the relevant person as far as i know.

**Comment by am6737** [comment]
> Hi am6737,
> 
> There is no automatic migration.
> 
> Only k8s itself manage its own resources and makes sure they are available even when node is down, kubevirt doesn't do as such.
> 
> Migration should be initiated by the relevant person as far as i know.

Thank you for your help.

Wishing you a wonderful day.

**Comment by oshoval** [comment]
Thanks, you too

**Comment by am6737** [comment]
> 

I'm sorry I opened this question again.

I still have some doubts as to why Kubevirt did not provide automatic migration scenarios

**Comment by simonyangcj** [comment]
hi am6737, I think it's hard to tell vm is really down or not. Let's say if we do have some solution to ping vm's ip in virt-controller.  It could be just a network issue between controller plane and host that vm running on if vm's ip is not response any more. We just cannot ensure vm is really down

**Comment by am6737** [comment]
> hi am6737, I think it's hard to tell vm is really down or not. Let's say if we do have some solution to ping vm's ip in virt-controller. It could be just a network issue between controller plane and host that vm running on if vm's ip is not response any more. We just cannot ensure vm is really down

Nice to receive your reply.

You are correct. For some reason, we cannot determine whether the virtual machine is actually in a shutdown state.

But I think if the node cannot be reached, even if the virtual machine is still running, it is no longer meaningful and cannot be connected through console, vnc, ssh, etc.

Shouldn't we try to migrate the virtual machine to a new healthy node to restore access to the virtual machine?

In summary, I want to ensure high availability of virtual machines.

What comes to my mind is the custom controller that detects node status and migrates virtual machines at the appropriate time.

**Comment by simonyangcj** [comment]
> But I think if the node cannot be reached, even if the virtual machine is still running, it is no longer meaningful and cannot be connected through console, vnc, ssh, etc.

Totally agree. I may be wrong, but I feel that  we might be able to do something but I don't think it should be triggered by kubevirt controller plane automatically.

**Comment by xpivarc** [comment]
Just to clarify. Kubevirt can spin up your VM on another node if you set the [runstrategy](https://kubevirt.io/user-guide/virtual_machines/run_strategies/). The only prerequisite is that something, external to Kubevirt, needs to fail the virt-launcer Pod, to prevent a split-brain situation.

There are already available operators such as Node Health Check and Self Node Remediation.

/close
Feel free to reopen if you have follow-up questions.
